### PR TITLE
[test] Add nvidia-smi check

### PR DESCRIPTION
**Pull request by victorusu**
This introduces the Nvidia SMI tests that check whether Accounting, Compute and ECC modes are set to the correct values.

### CODE REVIEW

**Comment by vkarak** [comment]
I am not sure that the location of the test is correct. Perhaps remove the `nvidia/` subfolder.

File: cscs-checks/system/nvidia/nvidia_smi_check.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    def __init__(self):
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = ['builtin']
+        self.executable = 'nvidia-smi'
+        self.num_tasks = 0
+        self.num_tasks_per_node = 1
+        self.exclusive = True
+        self.tags = {'nvidia', 'maintenance', 'production', 'single-node'}
+        self.maintainers = ['VH']
```
Comment (vkarak) [inline cscs-checks/system/nvidia/nvidia_smi_check.py:40]: Set all those in the class body and remove the `__init__()` altogether.

File: cscs-checks/system/nvidia/nvidia_smi_check.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+@rfm.simple_test
+class NVIDIASMIAccountCheck(NVIDIASMIBaseCheck):
```
Comment (vkarak) [inline cscs-checks/system/nvidia/nvidia_smi_check.py:44]: I would not use subclasses, but rather parameterize on the type of the test, e.g., `ACCOUNTING`, `ECC` etc.

File: cscs-checks/system/nvidia/nvidia_smi_check.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+def sanity_evaluation(regex, regex_group, output, refvalue, field_msg, num_nodes, nodelist):
```
Comment (vkarak) [inline cscs-checks/system/nvidia/nvidia_smi_check.py:10]: Most or all the logic here can go in a pre-sanity hook that would also set the `sanity_patterns`.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by vkarak** [comment]
In order to put everything to the class body, you need #1862.

**Comment by vkarak** [comment]
@victorusu I simplified and modernized the test. That's the new way! @jjotero Please confirm :-) We are also almost to the point of getting rid completely of `__init__()` in all cases. By the way, I don't why you were checking also the `Pending ECC Mode`. Did you have to? I am not checking it here, because I wanted to keep the test simpler.

File: cscs-checks/system/nvidia/nvidia_smi_check.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+class nvidia_smi_check(rfm.RunOnlyRegressionTest):
+    gpu_mode = parameter(['accounting', 'compute', 'ecc'])
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    valid_prog_environs = ['builtin']
+    executable = 'nvidia-smi'
+    executable_opts = ['-a', '-d']
+    num_tasks = 0
+    num_tasks_per_node = 1
+    exclusive = True
+    tags = {'maintenance', 'production'}
```
Comment (vkarak) [inline cscs-checks/system/nvidia/nvidia_smi_check.py:21]: @victorusu Flexible tests in maintenance and production? Nah...

Comment (vkarak) [reply cscs-checks/system/nvidia/nvidia_smi_check.py:21]: I would make it `diagnostic`.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/system/nvidia/nvidia_smi_check.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    maintainers = ['VH']
+    mode_values = variable(typ.Dict[str, str], value={
+        'accounting': 'Enabled',
+        'compute': 'Exclusive_Process',
+        'ecc': 'Enabled'
+    })
+
+    @rfm.run_before('run')
+    def set_variant_opts(self):
```
Comment (vkarak) [inline cscs-checks/system/nvidia/nvidia_smi_check.py:30]:
```suggestion
    def set_display_opt(self):
```

File: cscs-checks/system/nvidia/nvidia_smi_check.py (hunk 4 above)
Comment (victorusu) [reply cscs-checks/system/nvidia/nvidia_smi_check.py:21]: I talked internally with Matteo and he told me that 1 node is enough to test on maintenance, due to the use of the same image.

**Review by vkarak — APPROVED** [review]

**Review by jjotero — APPROVED** [review]
lgtm

**Review by teojgo — APPROVED** [review]

**Pull request merged**